rotkehlchen/db/search_assets.py

```diff
@@ -0,0 +1,119 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from polyleven import levenshtein
+
+from rotkehlchen.assets.types import AssetType
+from rotkehlchen.constants.assets import A_ETH, A_ETH2
+from rotkehlchen.constants.resolver import ChainID
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.db.filtering import AssetsFilterQuery
+
+
+def search_assets_levenshtein(
+        filter_query: 'AssetsFilterQuery',
+        db: 'DBHandler',
+        substring_search: str,
+        limit: Optional[int],
+        search_nfts: bool,
+) -> List[Dict[str, Any]]:
+    """Returns a list of asset details that match the search keyword using the Levenshtein distance approach."""  # noqa: E501
+    search_result = []
+    levenshtein_distances: List[int] = []
+    query, bindings = GlobalDBHandler()._prepare_search_assets_query(filter_query)
+    resolved_eth = A_ETH.resolve_to_crypto_asset()
+    with GlobalDBHandler().conn.read_ctx() as globaldb_cursor, db.conn.read_ctx() as cursor:
+        globaldb_cursor.execute(query, bindings)
+        treat_eth2_as_eth = db.get_settings(cursor).treat_eth2_as_eth
+        found_eth = False
+        for entry in globaldb_cursor:
+            # if an asset does not have a name and symbol, skip.
+            if entry[1] is None and entry[2] is None:
+                continue
+
+            substr_in_name = substr_in_symbol = None
+            # default to a value greater than the threshold to avoid false-positives
+            # and prevent raising `UnboundLocalError` when name/symbol is None.
+            lev_dist_name = lev_dist_symbol = 6
+            if entry[1] is not None:
+                name_casefold = entry[1].casefold()
+                lev_dist_name = levenshtein(substring_search, name_casefold)
+                substr_in_name = substring_search in name_casefold
+
+            if entry[2] is not None:
+                symbol_casefold = entry[2].casefold()
+                lev_dist_symbol = levenshtein(substring_search, symbol_casefold)
+                substr_in_symbol = substring_search in symbol_casefold
+
+            if not substr_in_name and not substr_in_symbol:
+                continue
```

Comment:
related to the comment of doing the non sensitive check I believe it can be totally be done in sql as it is always done unless there is something obvious that I'm missing. Also here we are calculating levenshtein distance but we might not need it as we just skip the result if `not substr_in_name and not substr_in_symbol` that doesn't depend on the distance